src/components/textInput/textInput.tsx

```diff
@@ -26,12 +29,106 @@ export interface TextInputProps {
   maxRows?: number
   /** Boolean that dictates whether `TextInput` takes up 100% width of the parent container. */
   fullWidth?: boolean
+  /** Boolean to allow option for speech-to-text. See which browsers are supported [here](https://developer.mozilla.org/en-US/docs/Web/API/SpeechRecognition#browser_compatibility). */
+  speechToText?: boolean
 }
 
 export default function TextInput(props: TextInputProps) {
   const [messageText, setMessageText] = useState<string>('')
+  const [errorMessage, setErrorMessage] = useState<string>('')
+  const [isRecording, setIsRecording] = useState(false)
+  const [isFocused, setIsFocused] = useState(false)
+  const [isEndingRecording, setIsEndingRecording] = useState(false)
 
   const isEmptyMessage = !messageText.trim().length
+  const speechToTextTooltipTitle = `${isRecording ? 'Stop' : 'Start'} speech to text`
+  const speechToTextInactiveColor = isFocused ? 'primary.main' : 'primary.light'
+  const speechToTextIconColor = isRecording
+    ? 'error.main'
+    : speechToTextInactiveColor
+  const speechToTextIconName = isRecording ? 'stop_circle' : 'speech_to_text'
+
+  const speechRecognitionErrors = {
+    'no-speech': 'no speech was detected',
+    aborted: 'speech input was aborted, possibly due to user action',
+    'audio-capture': 'audio capture failed',
+    network: 'network communication required for recognition failed',
+    'not-allowed':
+      'speech input disallowed due to security, privacy, or user preference',
+    'service-not-allowed':
+      'requested speech recognition service not allowed by user agent',
+    'bad-grammar': 'error in speech recognition grammar or unsupported format',
+    'language-not-supported':
+      'user agent does not support the specified language for recognition',
+  }
```

Comment:
Some of these are not errors, could we guide or suggest to try again with steps to resolve. For example, in case of service not allowed, I think users should be guided that they need to allow mic access when prompted. Similarly for no-speech, it could happen that user didn't say anything  For network error too, it means the internet connectivity was interrupted or is not there and user should check that before retrying.